internal/controller/nonadminbackupstoragelocation_controller_test.go

```diff
@@ -0,0 +1,466 @@
+
+			// Test case(s) to catch error when the corresponding secret is not in the cluster or it's not properly configured
+			if scenario.createNonAdminSecret && scenario.nonAdminSecretName != constant.EmptyString {
+				nonAdminSecret := buildTestNonAdminSecretForBsl(nonAdminBslNamespace, scenario.nonAdminSecretName, "testAccessKeyId", "testSecretAccessKey")
+				gomega.Expect(k8sClient.Create(ctx, nonAdminSecret)).To(gomega.Succeed())
+			}
+
+			k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+				Scheme: k8sClient.Scheme(),
+			})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			enforcedBslSpec := &velerov1.BackupStorageLocationSpec{}
+
+			if scenario.enforcedBslSpec != nil {
+				enforcedBslSpec = scenario.enforcedBslSpec
+			}
+			err = (&NonAdminBackupStorageLocationReconciler{
+				Client:          k8sManager.GetClient(),
+				Scheme:          k8sManager.GetScheme(),
+				OADPNamespace:   oadpNamespace,
+				EnforcedBslSpec: enforcedBslSpec,
+			}).SetupWithManager(k8sManager)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			go func() {
+				defer ginkgo.GinkgoRecover()
+				err = k8sManager.Start(ctx)
+				gomega.Expect(err).ToNot(gomega.HaveOccurred(), "failed to run manager")
+			}()
+			// wait manager start
+			managerStartTimeout := 10 * time.Second
+			pollInterval := 100 * time.Millisecond
+			ctxTimeout, cancel := context.WithTimeout(ctx, managerStartTimeout)
+			defer cancel()
+
+			err = wait.PollUntilContextTimeout(ctxTimeout, pollInterval, managerStartTimeout, true, func(ctx context.Context) (done bool, err error) {
+				select {
+				case <-ctx.Done():
+					return false, ctx.Err()
+				default:
+					// Check if the manager has started by verifying if the client is initialized
+					return k8sManager.GetClient() != nil, nil
+				}
+			})
+			// Check if the context timeout or another error occurred
+			gomega.Expect(err).ToNot(gomega.HaveOccurred(), "Manager failed to start within the timeout period")
+
+			ginkgo.By("Waiting Reconcile of create event")
+			nonAdminBsl := buildTestNonAdminBackupStorageLocation(nonAdminBslNamespace, nonAdminBslName, scenario.spec)
+			gomega.Expect(k8sClient.Create(ctxTimeout, nonAdminBsl)).To(gomega.Succeed())
+			// wait NonAdminBackupStorageLocation reconcile
+			time.Sleep(2 * time.Second)
+
+			ginkgo.By("Fetching NonAdminBackupStorageLocation after Reconcile")
+			gomega.Expect(k8sClient.Get(
+				ctxTimeout,
+				types.NamespacedName{
+					Name:      nonAdminBslName,
+					Namespace: nonAdminBslNamespace,
+				},
+				nonAdminBsl,
+			)).To(gomega.Succeed())
+
+			ginkgo.By("Validating NonAdminBackupStorageLocation Status")
+
+			gomega.Expect(checkTestNonAdminBackupStorageLocationStatus(nonAdminBsl, scenario.expectedStatus)).To(gomega.Succeed())
+
+			veleroBsl := &velerov1.BackupStorageLocation{}
+			if scenario.expectedStatus.VeleroBackupStorageLocation != nil && len(nonAdminBsl.Status.VeleroBackupStorageLocation.NACUUID) > 0 {
```

Comment:
this is not being run, need to add
```go
                                VeleroBackupStorageLocation: &nacv1alpha1.VeleroBackupStorageLocation{
					Status: nil,
				},
```
here https://github.com/mpryc/oadp-non-admin/blob/cc9b49bacce0286986c1bc5ef3ef5986ce06163b/internal/controller/nonadminbackupstoragelocation_controller_test.go#L438